tests/python/test_sql.py

```diff
@@ -121,6 +121,47 @@ def test_require(self):
         self._test("a12", [c4r12, c3r2, c1, c2],
                    dict(c1__c2__c4__c3=[c1, c2, c4r12, c3r2]))
 
+        self._test("a13", [c2, c7r2], dict(c2__c7=[c2, c7r2]))
+        self._test("a14", [c1, c2, c8r12], dict(c1__c2__c8=[c1, c2, c8r12]))
+        self._test("a15", [c9], dict(c9=[c9]))
+        self._test("a16", [c10], dict(c10=[c10]))
+        self._test("a17", [c11], dict(c11=[c11]))
+        self._test("a18", [c12], dict(c12=[c12]))
+
+    def _parse_reqs(self, reqs, expected_layers, expected_tables, expected_funcs):
```

Comment:
would be `_parse_props` not better instead `_parse_reqs` because you test here all properties not only requires